dev/tasks/tasks.yml

```diff
@@ -1321,13 +1321,13 @@ tasks:
       image: ubuntu-r-only-r
 
   # This also has -flto=auto
-  test-r-gcc-11:
+  test-r-gcc-12:
     ci: github
     template: docker-tests/github.linux.yml
     params:
       env:
         UBUNTU: 22.04
-        GCC_VERSION: 11
+        GCC_VERSION: 12
       # S3 support is not buildable with gcc11 right now
       flags: '-e ARROW_S3=OFF'
```

Comment:
Would you mind also trying to drop this line and see if it builds? I suspect it won't, but it's possible that the aws sdk + gcc build issues have been resolved by now (and we bumped the sdk version enough that we don't run into the old ones)